src/goto-programs/remove_skip.cpp

```diff
@@ -12,12 +12,22 @@ Author: Daniel Kroening, kroening@kroening.com
 #include "remove_skip.h"
 #include "goto_model.h"
 
-bool is_skip(const goto_programt &body, goto_programt::const_targett it)
+/// Determine whether the instruction is semantically equivalent to a skip
+/// (no-op).  This includes a skip, but also if(false) goto ..., goto next;
+///  next: ..., and (void)0.
+/// \param body  goto program containing the instruction
+/// \param it  instruction iterator that is tested for being a skip (or
+/// equivalent)
+/// \param ignore_labels  If the caller takes care of moving labels, then even
+/// skip statements carrying labels can be treated as skips (even though they
+/// may carry key information such as error labels).
+/// \return True, iff it is equivalent to a skip.
```

Comment:
I'm having a stuck record moment but ... what guarantees do our tools provide on the preservation of locations, instruction IDs, etc. through various transforms?  For example, if I am an entirely hypothetical commercial user of CPROVER, how do I map results back through transformation steps?

This is not an issue specifically or directly to do with this PR (although I think this PR may make it more involved) but I think it is something that does need to be considered at some point / in some way.